### DISCUSSION TITLE
devServer.hot: 'only' is not working.

### DISCUSSION THREAD

**Discussion by hamza761** [web-infra-dev/rspack#3944] [Q&A]
<img width="885" alt="Screenshot 2023-08-13 at 12 10 08 AM" src="https://github.com/web-infra-dev/rspack/assets/66865958/fb05a59e-ffc0-4727-98fb-bd1d1b5dbb53">
As evident from the preceding screenshot, the devServer.hot configuration exclusively accommodates boolean values, disregarding the 'only' option. Despite webpack's provision of the 'only' option, my aim is to attain Hot Module Replacement (HMR) functionality without triggering complete page reloads. 
Upon encountering an error and subsequently resolving it, the current behavior involves an automatic full page reload. However, my preference is to prevent this automatic reload from happening. I would prefer to receive a notification indicating the need for a full page reload, allowing me to trigger the reload manually. During this manual reload, I intend to re-render the page. Importantly, I wish to maintain the functioning of Hot Module Replacement (HMR) in the same manner as it operates when no errors are present. To achieve this, I am aware that webpack's devServer.hot can be set to 'only', effectively halting the automatic full page reloads following error resolution.
Could you kindly guide me on achieving this objective in RSPACK? Thank you.

**Comment by hardfist** [comment]
we're refactoring devServer.hot and will make it behave the same as webpack in the future

**Reply by hamza761** [reply]
It's great to hear. Looking forward to seeing the improvements! 👍. Thank
